internal/fs/vss_windows.go

```diff
@@ -0,0 +1,977 @@
+func (vss *IVssBackupComponents) IsVolumeSupported(volumeName string) (bool, error) {
+	volumeNamePointer, err := syscall.UTF16PtrFromString(volumeName)
+	if err != nil {
+		panic(err)
+	}
+
+	var isSupportedRaw uint32
+	var result uintptr
+
+	if runtime.GOARCH == "386" {
+		id := (*[4]uintptr)(unsafe.Pointer(ole.IID_NULL))
+
+		result, _, _ = syscall.Syscall9(vss.getVTable().isVolumeSupported, 7,
+			uintptr(unsafe.Pointer(vss)), id[0], id[1], id[2], id[3],
+			uintptr(unsafe.Pointer(volumeNamePointer)), uintptr(unsafe.Pointer(&isSupportedRaw)), 0,
+			0)
+	} else {
+		result, _, _ = syscall.Syscall6(vss.getVTable().isVolumeSupported, 4,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(ole.IID_NULL)),
+			uintptr(unsafe.Pointer(volumeNamePointer)), uintptr(unsafe.Pointer(&isSupportedRaw)), 0,
+			0)
+	}
+
+	var isSupported bool
+	if isSupportedRaw == 0 {
+		isSupported = false
+	} else {
+		isSupported = true
+	}
+
+	return isSupported, newVssErrorIfResultNotOK("IsVolumeSupported() failed", HRESULT(result))
+}
+
+// StartSnapshotSet calls the equivalent VSS api.
+func (vss *IVssBackupComponents) StartSnapshotSet() (ole.GUID, error) {
+	var snapshotSetID ole.GUID
+	result, _, _ := syscall.Syscall(vss.getVTable().startSnapshotSet, 2,
+		uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&snapshotSetID)), 0,
+	)
+
+	return snapshotSetID, newVssErrorIfResultNotOK("StartSnapshotSet() failed", HRESULT(result))
+}
+
+// AddToSnapshotSet calls the equivalent VSS api.
+func (vss *IVssBackupComponents) AddToSnapshotSet(volumeName string, idSnapshot *ole.GUID) error {
+	volumeNamePointer, err := syscall.UTF16PtrFromString(volumeName)
+	if err != nil {
+		panic(err)
+	}
+
+	var result uintptr = 0
+
+	if runtime.GOARCH == "386" {
+		id := (*[4]uintptr)(unsafe.Pointer(ole.IID_NULL))
+
+		result, _, _ = syscall.Syscall9(vss.getVTable().addToSnapshotSet, 7,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(volumeNamePointer)), id[0], id[1],
+			id[2], id[3], uintptr(unsafe.Pointer(idSnapshot)), 0, 0)
+	} else {
+		result, _, _ = syscall.Syscall6(vss.getVTable().addToSnapshotSet, 4,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(volumeNamePointer)),
+			uintptr(unsafe.Pointer(ole.IID_NULL)), uintptr(unsafe.Pointer(idSnapshot)), 0, 0)
+	}
+
+	return newVssErrorIfResultNotOK("AddToSnapshotSet() failed", HRESULT(result))
+}
+
+// PrepareForBackup calls the equivalent VSS api.
+func (vss *IVssBackupComponents) PrepareForBackup() (*IVSSAsync, error) {
+	var oleIUnknown *ole.IUnknown
+	result, _, _ := syscall.Syscall(vss.getVTable().prepareForBackup, 2,
+		uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&oleIUnknown)), 0)
+
+	err := newVssErrorIfResultNotOK("PrepareForBackup() failed", HRESULT(result))
+	return vss.convertToVSSAsync(oleIUnknown, err)
+}
+
+// apiBoolToInt converts a bool for use calling the VSS api
+func apiBoolToInt(input bool) uint {
+	if input {
+		return 0xffff
+	}
```

Comment:
I think `1` should be safer here: `bSelectComponents` and other defined as `bool`, not `BOOL` and it is known that some parts of Windows erroneously check for equality (`v == true`). I have personally encountered this with audio API: IAudioEndpointVolume.SetMute(1) and IAudioEndpointVolume.SetMute(255) lead to different results on some systems.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/c4792c69fa5f5b7f4bf40915bbcda1592301bbea